rules-reviewed/jboss-eap/websphere/websphere-mq.windup.xml

```diff
@@ -0,0 +1,846 @@
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+
+
+
+        <rule id="websphere-mq-05000">
+            <when>
+                <javaclass references="com.ibm.msg.client.wmq.WMQConstants"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ Everyplace JMS constants">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"/>
+                </classification>
+                <hint effort="1">
+                    <message>Replace with Java EE 6 JMS.</message>
+                    <link description="Java EE 6 JMS Tutorial" href="https://docs.oracle.com/javaee/6/tutorial/doc/bnceh.html"/>
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="websphere-mq-06000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeJMSAuthenticator"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ Everyplace JMS authentication interface">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"/>
+                </classification>
+                <hint effort="1">
+                    <message>Replace with Java EE 6 JAAS for JMS authentication.</message>
+                    <link description="JBoss EAP security domain" href="https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.2/html-single/Development_Guide/index.html#About_Security_Domains"/>
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="websphere-mq-07000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeJMSMsgFieldNames"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQe constants" description="Defines a number of String constants that are used to name fields when a JMS message is mapped to an MQeMsgObject.">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"/>
+                </classification>
+            </perform>
+        </rule>
+
+
+        <rule id="websphere-mq-08000">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.MQeConnection"/>
+            </when>
+            <perform>
+                <hint effort="1">
+                    <message>A more general, and recommended, approach is to use the Java EE 6 JMS class `javax.jms.Connection`.</message>
```

Comment:
This hint needs to be a little clearer. First, the approach is required, not recommended. Second, an explanation of the problem is also useful to help people understand what is going on. Such as my earlier comments on other hints that need a little more explanation.